src/api/modules/sponsor/setEdition.ts

```diff
@@ -13,9 +15,83 @@ const promptSwitchToAccount = async (account: string, initial: boolean) => {
     : `Only current account sponsor (${chalk.blue(account)}) can change its edition`
   const proceed = await promptConfirm(`${reason}. Do you want to switch to account ${chalk.blue(account)}?`)
   if (!proceed) {
-    return
+    return false
   }
   await switchAccount(account, {})
+  return true
+}
+
+const confirmAndSwitchEnvironment = async (sponsorAccount: string, targetAccount: string, targetWorkspace: string) => {
+  if (!sponsorAccount) {
+    if (targetWorkspace !== 'master') {
+      throw createFlowIssueError('Can only set initial edition in master workspace')
+    }
+    return await promptSwitchToAccount('vtex', true)
+  }
+
+  const workspaceIsOk = targetWorkspace !== 'master' || await promptWorkspaceMaster(targetAccount)
+  return workspaceIsOk && await promptSwitchToAccount(sponsorAccount, false)
+}
+
+const tenantProvisionerApp = 'vtex.tenant-provisioner'
+
+const promptInstallTenantProvisioner = async (account: string) => {
+  log.warn(`Tenant provisioner app seems not to be installed in sponsor account ${chalk.blue(account)}.`)
+  const proceed = await promptConfirm(`Do you want to install ${chalk.blue(tenantProvisionerApp)} in account ${chalk.blue(account)} now?`)
+  if (!proceed) {
+    return false
+  }
+  await appsInstall([tenantProvisionerApp], { force: false })
+  return true
+}
+
+const isProvisionerNotInstalledError = (err: any) => {
+  const res = err.response
+  return res?.status === 404 &&
+      res.data?.source === 'Vtex.Kube.Router' &&
+      res.data?.code === 'NotFound'
+}
+
+const trySetEditionOnce = async (client: Sponsor, targetAccount: string, targetWorkspace: string, edition: string) => {
+  try {
+    await client.setEdition(targetAccount, targetWorkspace, edition)
+    return true
+  } catch (err) {
+    if (!isProvisionerNotInstalledError(err)) {
+      throw err
+    }
+    return false
+  }
+}
+
+const maxSetEditionRetries = 3
+
+const sleepSec = (sec: number) => new Promise(resolve => setTimeout(resolve, sec * 1000))
```

Comment:
_non blocking_
This seems to be a very generic function. What do you think about moving it to another file, with more generic components? It could be a `utils` file for instance